pkg/controller/datavolume/controller-base.go

```diff
@@ -112,6 +114,22 @@ func pvcIsPopulated(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) boo
 	return ok && dvName == dv.Name
 }
 
+func checkStaticProvisionPending(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) bool {
+	if pvc == nil || dv == nil {
+		return false
+	}
+	if _, ok := dv.Annotations[cc.AnnCheckStaticVolume]; !ok {
+		return false
+	}
+	_, ok := pvc.Annotations[cc.AnnPersistentVolumeList]
+	return ok
+}
+
+func shouldSetPending(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) bool {
+	return checkStaticProvisionPending(pvc, dv) ||
+		(pvc == nil && dv.Status.Phase == cdiv1.PhaseUnset)
```

Comment:
I would add `&& dv != nil`

Reply:
dv will never be nil in either case and `shouldSetPending` kind of assumes the dv exists